source/databricks/calculation_engine/package/calculation/preparation/transformations/basis_data_factory.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2020 Energinet DataHub A/S
+#
+# Licensed under the Apache License, Version 2.0 (the "License2");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import col
+
+from package.calculation.CalculationResults import BasisDataContainer
+from package.calculation.preparation.transformations import basis_data
+from package.constants import BasisDataColname, PartitionKeyName
+from package.infrastructure import logging_configuration
+
+
+def create(
+    metering_point_periods_df: DataFrame,
+    metering_point_time_series_df: DataFrame,
+    time_zone: str,
+) -> BasisDataContainer:
+    basis_data_container = BasisDataContainer()
+
+    with logging_configuration.start_span("basis_data_prepare"):
+        (
+            timeseries_quarter_df,
+            timeseries_hour_df,
+        ) = basis_data.get_metering_point_time_series_basis_data_dfs(
+            metering_point_time_series_df, time_zone
+        )
+
+        master_basis_data_df = basis_data.get_master_basis_data_df(
+            metering_point_periods_df
+        )
+
+        # Get basis data for energy suppliers
+        (
+            master_basis_data_df,
```

Comment:
The `_df`  suffix convention is deprecated. We used it before we adapted good habits with type annotations.
This suggestion applies to multiple names in the PR.
```suggestion
            master_basis_data,
```

Reply:
Done